Eevee.Sleep.Bot/Modules/SlashCommands/ChangeRoleSlashModule.cs

```diff
@@ -0,0 +1,58 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for role things.")]
+public class ChangeRoleSlashModule : InteractionModuleBase<SocketInteractionContext> {
+    [SlashCommand("change", "Select a specific one from the role you have")]
+    [UsedImplicitly]
+    public async Task ChangeRoleAsync() {
+        var roles = UserRoleContoller.FindRoleIdsByUserId(Context.User.Id.ToString());
+        var builder = new ComponentBuilder();
+
+        foreach(var role in roles) {
+            builder.WithButton(
+                label: Context.Guild.Roles.Where(x => x.Id == role).Select(x => x.Name).First(), 
+                customId: $"role{role}"
```

Comment:
Use the role info serializer mentioned in the other comments for serializing this info

Reply:
Already answered here
https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25#discussion_r1609964437